### PR TITLE
[DOC] Add sort processor documentation

### PR DESCRIPTION
**Pull request by vagimeli**
### Description
Closes content gap

### Issues Resolved
#4647 

### Checklist
- [ x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@hdhalter We need a dev to take the first pass at drafting this content. I can support them in refining the content and getting it through the documentation process. The ingest processor template is provided with this PR, so it should help the dev get started. Thanks!

**Comment by vagimeli** [comment]
@gaobinlong This PR is ready for your review at your availability. Thank you for all your technical reviews on these topics!

File: _ingest-pipelines/processors/sort.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
+---
+layout: default
+title: Sort
+parent: Ingest processors
+nav_order: 250
+---
+
+# Sort processor
+
+The `sort` processor sorts an array of items in either ascending or descending order. The processor takes in a field name (`field`) and a sort order (`order`), and it creates a new field (`target_field`) with the sorted array.
```
Comment (vagimeli) [inline _ingest-pipelines/processors/sort.md:10]:
```suggestion
The `sort` processor sorts an array of items in either ascending or descending order. Numeric arrays are sorted numerically, while string or mixed arrays (strings and numbers) are sorted lexicographically. Throws an error if the input is not an array.
```

Comment (vagimeli) [reply _ingest-pipelines/processors/sort.md:10]: @gaobinlong Please confirm this change addresses your comments accurately. Thank you.

File: _ingest-pipelines/processors/sort.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `sort` processor.
+
+| Parameter  | Required/Optional  | Description  |
+|---|---|---|
+`field`  | Required | The field to be sorted.
```
Comment (vagimeli) [inline _ingest-pipelines/processors/sort.md:35]:
```suggestion
`field`  | Required | The field to be sorted.
```
```suggestion
`field`  | Required | The field to be sorted. Must be an array.
```

File: _ingest-pipelines/processors/sort.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `sort` processor.
+
+| Parameter  | Required/Optional  | Description  |
+|---|---|---|
+`field`  | Required | The field to be sorted. Must be an array.
+`order`  | Required | The sort order to apply. Accepts `asc` for ascending or `desc` for descending. Default is `asc`.
```
Comment (vagimeli) [inline _ingest-pipelines/processors/sort.md:36]:
```suggestion
`order`  | Optional | The sort order to apply. Accepts `asc` for ascending or `desc` for descending. Default is `asc`.
```

**Comment by vagimeli** [comment]
@natebower Thank you for the thorough review. I've accepted your edits. Will you approve for merging? This is part of the series that I opened PRs and co-authored with dev, but can't approve. Thank you.

File: _ingest-pipelines/processors/sort.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
+---
+layout: default
+title: Sort
+parent: Ingest processors
+nav_order: 250
+---
+
+# Sort processor
+
+The `sort` processor sorts an array of items in either ascending or descending order. Numeric arrays are sorted numerically, while string or mixed arrays (strings and numbers) are sorted lexicographically. Throws an error if the input is not an array.
```
Comment (natebower) [inline _ingest-pipelines/processors/sort.md:10]:
```suggestion
The `sort` processor sorts an array of items in either ascending or descending order. Numeric arrays are sorted numerically, while string or mixed arrays (strings and numbers) are sorted lexicographically. The processor throws an error if the input is not an array.
```

File: _ingest-pipelines/processors/sort.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `sort` processor.
+
+| Parameter  | Required/Optional  | Description  |
+|---|---|---|
+`field`  | Required | The field to be sorted. Must be an array.
+`order`  | Optional | The sort order to apply. Accepts `asc` for ascending or `desc` for descending. Default is `asc`.
+`target_field` | Optional | The name of the field where the sorted array is stored. If not specified, the sorted array is stored in the same field as the original array (the `field` variable).
```
Comment (natebower) [inline _ingest-pipelines/processors/sort.md:37]:
```suggestion
`target_field` | Optional | The name of the field in which the sorted array is stored. If not specified, then the sorted array is stored in the same field as the original array (the `field` variable). 
```

File: _ingest-pipelines/processors/sort.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+POST testindex1/_doc?pipeline=sort-pipeline
+{
+  "my_array_field": [3, 1, 4, 1, 5, 9, 2, 6, 5]
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The request indexes the document into the index `testindex1` and then indexes and then indexes all documents with the `my_array_field` sorted in descending order, as shown in the following response:
```
Comment (natebower) [inline _ingest-pipelines/processors/sort.md:149]:
```suggestion
The request indexes the document into the index `testindex1` and then indexes all documents with the `my_array_field` sorted in descending order, as shown in the following response:
```

**Review by natebower — APPROVED** [review]
@vagimeli LGTM with a few minor changes. Thanks!

**Pull request merged**